WcaOnRails/app/models/wcif.rb

```diff
@@ -13,6 +13,23 @@ def json
     {
       "formatVersion" => "1.0",
       "id" => competition.id,
+      "events" => competition.competition_events.map(&method(:competition_event_to_json)),
+    }
+  end
+
+  private def competition_event_to_json(competition_event)
```

Comment:
I think the flaw in this argument is in assuming that the structure of the WCIF will mirror the structure of the tables/classes in our RoR app. That said, in this particular place, it would work, so I'll give it a shot. I suspect this pattern won't work for @viroulep when he gets started on adding registration data to the WCIF.